fakeserver/server_test.go

```diff
@@ -150,3 +155,40 @@ func TestPersistAssignment(t *testing.T) {
 		require.Equal(t, "control", (*assignments)["test.test_experiment"])
 	})
 }
+
+func TestPersistAssignmentV2(t *testing.T) {
```

Comment:
@smudge i snuck a test in here too since i noticed that there was one for the old way